src/k2/cmd/tpcc/transactions.h

```diff
@@ -800,3 +800,374 @@ class OrderStatusT : public TPCCTxn
+                future<> delNORow_f = delNewOrderRow(idx);
+
+                // when we got NO_O_ID, update the carrierID in ORDER table
+                future<> updateCarrierID_f = updateCarrierID(idx);
+
+                // when we got NO_O_ID, retrieve the "customerID" and "orderLineCount" in ORDER table
+                future<> updates_f = getCustomerID(idx)
+                .then([this, idx] {
+                    // (1) when we got "orderLineCount", all rows in the ORDER_LINE table with mathing w_id, d_id and
+                    //     o_id are seclected, get the sum of all OL_AMOUNT;
+                    // (2) when we got "sum of all OL_AMOUNT" and "customerID", update the row in the CUSTOMER table.
+                    future<> updateCustomerRow_f = getOLSumAmount(idx)
+                    .then([this, idx] {
+                        return updateCustomerRow(idx);
+                    });
+
+                    // when we got "orderLineCount", update all the "DeliveryDates" in ORDER LINE table
+                    // with matching w_id, d_id and o_id.
+                    future<> updateOLDeliveryD_f = updateOLDeliveryD(idx);
+
+                    return when_all_succeed(std::move(updateOLDeliveryD_f), std::move(updateCustomerRow_f))
+                    .then_wrapped([this] (auto&& fut) {
+                        if (fut.failed()) {
+                            _failed = true;
+                        }
+                        return make_ready_future();
+                    });
+                });
+
+                return when_all_succeed(std::move(updates_f), std::move(updateCarrierID_f), std::move(delNORow_f))
+                .then_wrapped([this, idx] (auto&& fut) {
+                    if (fut.failed()) {
+                        _failed = true;
+                    }
+                    return make_ready_future();
+                });
+            });
+        });
+    }
+
+    // get lowest NO_O_ID row in NEW-ORDER table with matching _w_id and _d_id
+    future<bool> getOrderID(uint32_t idx) {
+        return _client.createQuery(tpccCollectionName, "neworder")
+        .then([this, idx](auto&& response) mutable {
+            if (!response.status.is2xxOK()) {
+                K2DEBUG("DeliveryT getOrderID createQuery failed.");
+                _failed = true;
+                return make_ready_future<bool>(false);
+            }
+
+            return  do_with(std::move(response.query), [this, idx] (Query& query_oid) {
+                // make Query request and set Query rules.
+                // 1) just return the lowest SKVRecord; 2) add Projection fields (i.e. oid) as needed;
+                query_oid.startScanRecord.serializeNext<int16_t>(_w_id);
+                query_oid.startScanRecord.serializeNext<int16_t>(_d_id[idx]);
+                query_oid.endScanRecord.serializeNext<int16_t>(_w_id);
+                query_oid.endScanRecord.serializeNext<int16_t>(_d_id[idx] + 1);
+                query_oid.setLimit(1);
```

Comment:
It would be more efficient if we could do the whole batch size in one query. Up to you if you want to change it for this PR or not.